docs/DesignDocs/PluginArchitecture.md

```diff
@@ -0,0 +1,65 @@
+Plugin Architecture
+=====
+
+This document describes the principles behind the openenclave plugin
+architecture and establishes the framework for Third-party Testing of those
+plugins.
+
+
+Motivation
+-----
+
+Set and communicate clearly our "north star" for enabling a community-driven
+development model, creating a strrong abstraction layer to support
+standardization across all enclaves while balancing the need to innovate
+quickly and adapt to new developments in hardware.
+
+
+User Experience
+------
+
+Any developer wishing to contribute a new plugin should read this document to understand
+the overall principles behind our plugin architecture, and how to integrate testing
+of their plugin to build community confidence in the code quality.
+
+Specific APIs will be defined in other documents.
+
+
+Specification
+------
+
+In the openenclave/plugin directory, a set of stubs (emulators) will be
+implemented, which facilitate testing of as much of the in-tree code as
+possible without relying on a physical hardware TEE. Virtual TEEs may be
+employed, and other forms of emulation may be considered and added in the
+future.
+
+This "stub" also serves as a template for anyone to create a hardware-specific
+plugin. As such, the "stub" code should be of the highest quality. It should
+implement all function calls, and demonstrate the expected coding standards
+which the openenclave community holds all contributors to.
+
+Hardware-specific code should be contributed within an out of tree plugin.
+
+Additionally, hardware-specific plugins should be tested in third-party CI/CD
+environment which subscribes to GitHub change notifications and posts the
```

Comment:
grammar: s/environment which subscribes/environments that subscribe/ (plural, since "plugins" is plural).
Also s/its/their/ in line 46